backend/wmg/data/snapshot.py

```diff
@@ -103,100 +103,103 @@ def load_snapshot(
 
-    if read_versioned_snapshot:
-        return data_schema_dir_path
-    return WMG_ROOT_DIR_PATH
+    return data_schema_dir_rel_path
 
 
-def _get_wmg_snapshot_dir_path(*, snapshot_schema_version: str, snapshot_id: str, read_versioned_snapshot: bool) -> str:
+def _get_wmg_snapshot_rel_path(snapshot_schema_version: str, snapshot_id: str) -> str:
     """
-    Get path to the snapshot id directory for a the given snapshot schema version.
+    Get relative path to the snapshot id directory for a the given snapshot schema version.
     """
-    data_schema_dir_path = _get_wmg_snapshot_schema_dir_path(
-        snapshot_schema_version=snapshot_schema_version,
-        read_versioned_snapshot=read_versioned_snapshot,
-    )
+    data_schema_dir_rel_path = _get_wmg_snapshot_schema_dir_rel_path(snapshot_schema_version)
 
-    snapshot_id_dir_path = f"{data_schema_dir_path}/{snapshot_id}" if data_schema_dir_path else snapshot_id
-    return snapshot_id_dir_path
+    snapshot_id_dir_rel_path = f"{data_schema_dir_rel_path}/{snapshot_id}"
+    return snapshot_id_dir_rel_path
 
 
-def _get_wmg_snapshot_dir_s3_uri(snapshot_dir_path: str) -> str:
+def _get_wmg_snapshot_fullpath(snapshot_rel_path: str, snapshot_local_disk_path: Optional[str] = None) -> str:
     """
-    Get the s3 uri of the snapshot directory
+    Return the full path of the snapshot on local disk or S3 URI of the snapshot.
     """
-    wmg_config = WmgConfig()
 
-    return os.path.join("s3://", wmg_config.bucket, snapshot_dir_path)
+    if snapshot_local_disk_path:
+        return os.path.join(snapshot_local_disk_path, snapshot_rel_path)
 
+    wmg_config = WmgConfig()
+    return os.path.join("s3://", wmg_config.bucket, snapshot_rel_path)
 
-def _load_snapshot(*, snapshot_schema_version: str, snapshot_id: str, read_versioned_snapshot: bool) -> WmgSnapshot:
-    snapshot_dir_path = _get_wmg_snapshot_dir_path(
-        snapshot_schema_version=snapshot_schema_version,
-        snapshot_id=snapshot_id,
-        read_versioned_snapshot=read_versioned_snapshot,
-    )
 
-    cell_type_orderings = _load_cell_type_order(snapshot_dir_path)
-    primary_filter_dimensions = _load_primary_filter_data(snapshot_dir_path)
-    filter_relationships = _load_filter_graph_data(snapshot_dir_path)
-    cell_type_ancestors = _load_cell_type_ancestors(snapshot_dir_path)
-    dataset_metadata = _load_dataset_metadata(snapshot_dir_path)
+def _load_snapshot(
+    *, snapshot_schema_version: str, snapshot_id: str, snapshot_local_disk_path: Optional[str] = None
+) -> WmgSnapshot:
+    snapshot_rel_path = _get_wmg_snapshot_rel_path(snapshot_schema_version, snapshot_id)
 
-    snapshot_base_uri = _get_wmg_snapshot_dir_s3_uri(snapshot_dir_path)
-    logger.info(f"Loading WMG snapshot from directory path {snapshot_dir_path} into URI {snapshot_base_uri}")
+    cell_type_orderings = _load_cell_type_order(snapshot_rel_path, snapshot_local_disk_path)
```

Comment:
Can we add some more comments explaining _rel_path vs local_disk_path vs full_path and how these values logically interact with each other? If this already exists, apologies in advance.

Reply:
@atarashansky - I will add that and change the naming to be clearer. I agree that this can be confusing